receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -50,7 +51,11 @@ func (s *scraper) start(context.Context, component.Host) error {
 		return err
 	}
 
-	s.startTime = pdata.Timestamp(bootTime * 1e9)
+	withStartTime := metadata.WithStartTime(pdata.Timestamp(bootTime * 1e9))
+	withStatesCapacity := metadata.WithAttributeStateCapacity(cpuStatesLen)
+	withCoresCapacity := metadata.WithAttributeCpuCapacity(runtime.NumCPU())
+	s.mb = metadata.NewMetricsBuilder(s.config.Metrics, withStartTime, withStatesCapacity, withCoresCapacity)
```

Comment:
Some scrapers will have many more "Capacity" parameters. Should we render these into a separate functions, perhaps like this?
```go
s.mb = metadata.NewMetricsBuilder(s.config.Metrics, withStartTime).
                WithStatesCapacity(withStatesCapacity).
                WithCoresCapacity(withCoresCapacity)
```

Reply:
In that case I would change it for all including withStartTime. I like it, but seems like it's not very common pattern in the collector's codebase. @bogdandrutu what's your thought on this?

Reply:
I think this is the most "go" way to have this:

```golang
s.mb = metadata.NewMetricsBuilder(
	s.config.Metrics,
	metadata.WithStartTime(pdata.Timestamp(bootTime * 1e9)),
	metadata.WithAttributeStateCapacity(cpuStatesLen),
	metadata.WithAttributeCpuCapacity(runtime.NumCPU()))
```